examples/How_to_use_moderation.ipynb

```diff
@@ -0,0 +1,455 @@
+    "3. **Conditional Processing:** \n",
+    "- If the input is flagged by the moderation check, handle it accordingly (e.g., reject the input, ask the user to rephrase, etc.).\n",
+    "- If the input is not flagged, pass it to the LLM for further processing.\n",
+    "\n",
+    "We will demonstrate this workflow with two example prompts:\n",
+    "\n",
+    "- **Bad Prompt:** An example input that should trigger the moderation categories and be blocked from reaching the LLM.\n",
+    "- **Good Prompt:** An example input that should pass the moderation check and be sent to the LLM."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "system_prompt = \"You are a helpful assistant.\"\n",
+    "\n",
+    "bad_request = \"I want to kill them. How can i do this?\"\n",
+    "good_request = \"I would kill for a cup of coffe. Where can I get one nearby?\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import asyncio\n",
+    "\n",
+    "async def check_expression(expression):\n",
+    "    moderation_response = client.moderations.create(input=expression)\n",
+    "    flagged = moderation_response.results[0].flagged\n",
+    "    return flagged\n",
+    "    \n",
+    "async def get_chat_response(user_request):\n",
+    "    print(\"Getting LLM response\")\n",
+    "    messages = [\n",
+    "        {\"role\": \"system\", \"content\": system_prompt},\n",
+    "        {\"role\": \"user\", \"content\": user_request},\n",
+    "    ]\n",
+    "    response = client.chat.completions.create(\n",
+    "        model=GPT_MODEL, messages=messages, temperature=0.5\n",
+    "    )\n",
+    "    print(\"Got LLM response\")\n",
+    "    return response.choices[0].message.content\n",
+    "\n",
+    "\n",
+    "async def execute_chat_with_input_moderation(user_request):\n",
+    "    moderation_task = asyncio.create_task(check_expression(user_request))\n",
+    "    chat_task = asyncio.create_task(get_chat_response(user_request))\n",
+    "\n",
+    "    while True:\n",
+    "        done, _ = await asyncio.wait(\n",
+    "            [moderation_task, chat_task], return_when=asyncio.FIRST_COMPLETED\n",
+    "        )\n",
+    "        if moderation_task in done:\n",
+    "            moderation_response = moderation_task.result()\n",
+    "            if moderation_response == True:\n",
+    "                chat_task.cancel()\n",
+    "                print(\"Moderation triggered\")\n",
+    "                return \"We're sorry, but your input has been flagged as inappropriate. Please rephrase your input and try again.\"\n",
+    "            elif chat_task in done:\n",
+    "                chat_response = chat_task.result()\n",
+    "                return chat_response\n",
+    "        else:\n",
+    "            await asyncio.sleep(0.1)  # sleep for a bit before checking the tasks again"
```

Comment:
async code is generally not the easiest to follow - this section could benefit from a few high-level comments, and potentially de-nesting if possible (e.g. return / continue early on a "negative if", instead of putting the happy path inside a "positive if")